src/raftstore/store/fsm/peer.rs

```diff
@@ -1338,6 +1338,9 @@ impl<'a, T: Transport, C: PdClient> PeerFsmDelegate<'a, T, C> {
         // We can't destroy a peer which is applying snapshot.
         assert!(!self.fsm.peer.is_applying_snapshot());
 
+        // Mark itself as pending_remove
+        self.fsm.peer.pending_remove = true;
```

Comment:
Why?

Reply:
Because this PR also cherry-pick #7492

Reply:
I recall master has changed the destroy flow and 3.0 doesn't have them, will it be a problem?